### DISCUSSION TITLE
Questions about dynamic routes

### DISCUSSION THREAD

**Discussion by luknl** [vercel/next.js#15832] [Help] [answered]
Hi there!

On our website, we have an object called "project". You can create one, view one and edit one. 
What we would like is those routes to match the same file: `/project/1` and `/project/1/foo`. With `/project/1/edit` going to a different route, and `/project/create` still working.
The current arborescence is like this:
```
project/
* [id]/ 
* - index.js
* - edit.js
* create.js
```
Is it possible?

I tried with another, simpler object we have, and I made it work with `/need/1` and `/need/1/foo` working and matching the same file. Arborescence I used for this one was:
```
need/
* […id]/index.js
```

Thank you!

**Comment by Timer** [comment, marked as answer]
Sure, the structure you showed above works fine!

```
project/
* [id]/ 
* - index.js
* - edit.js
* create.js
```

This will always match `/project/create` before `/project/[id]`.

**Reply by luknl** [reply]
With this structure, when I try to access `/project/1/foo`, it gives me a 404 page.

**Reply by Timer** [reply]
Oh, you'll need to create `pages/project/[id]/[action].js` or [use a rewrite](https://nextjs.org/docs/api-reference/next.config.js/rewrites).

**Reply by luknl** [reply]
Actually I tried with `/project/[id]/[...index].js` and `/project/[id]/[index].js`. They both made `/project/1` go to a 404 page, but `/project/1/foo` working.
Then I tried with the new [catch all routes feature](https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes), so with `/project/[id]/[[...index]].js`, and it now works 🎉
Just changed this, the structure of the other files stayed the same.
